jaeger-core/src/main/java/com/uber/jaeger/samplers/RemoteControlledSampler.java

```diff
@@ -75,47 +77,58 @@ public ReentrantReadWriteLock getLock() {
     return lock;
   }
 
-  private void updateSampler() {
+  /**
+   * Updates {@link #sampler} to a new sampler when it is different.
+   */
+  private synchronized void updateSampler() {
```

Comment:
this is way too coarse, you're going to block creation of all traces in the process while you're making a network call to the agent